src/components/structuredefSettings/StructuredefSettings.tsx

```diff
@@ -1,70 +1,297 @@
-      dispatch(getStructureDefSuccess(profileToEdit));
+  // CHANGE STATE
+  const handleEditSettings = (
+    e: React.MouseEvent<HTMLInputElement, globalThis.MouseEvent>
+  ) => {
+    if (structureDefinition) {
+      e.preventDefault();
+      // content required
+      structureDefinition.url = url as Uri.AsObject;
+      structureDefinition.name = name as String.AsObject;
+      structureDefinition.kind = kind as StructureDefinition.KindCode.AsObject;
+      structureDefinition.abstract = abstract as Boolean.AsObject;
+      structureDefinition.status = status as StructureDefinition.StatusCode.AsObject;
+      // content not required
+      structureDefinition.date = (date as unknown) as DateTime.AsObject;
+      createId(structureDefinition, id as Id.AsObject);
+      createPublisher(structureDefinition, publisher as String.AsObject);
+      createDescription(structureDefinition, description as String.AsObject);
+      createPurpose(structureDefinition, purpose);
+      createExperimental(structureDefinition, experimental);
+      createTitle(structureDefinition, title);
+      createCopyright(structureDefinition, copyright);
+      dispatch(getStructureDefSuccess(structureDefinition));
     }
   };
 
-  return (
+  const renderContact = (
     <>
-      <h2>Structure definition Settings</h2>
-      <label htmlFor="name">Name of profile</label>
-      <input type="text" name="name" onChange={(e) => editName(e)} />
+      <label htmlFor="contactName">Contact Name</label>
+      <input type="text" name="contactName" />
       <br />
-      <label htmlFor="name">Description</label>
-      <input type="text" name="name" onChange={(e) => editDescription(e)} />
+      <select name="contactSystem">
+        <option value="phone">Phone</option>
+        <option value="fax">fax</option>
+        <option value="email">email</option>
+        <option value="pager">pager</option>
+        <option value="url">url</option>
+        <option value="sms">sms</option>
+        <option value="other">other</option>
+      </select>
       <br />
-      <input type="datetime-local" onChange={(e) => editTime(e)} />
-      <input
-        type="button"
-        value="submit"
-        onClick={() => handleEditSettings()}
-      />
+      <label htmlFor="contactValue">Value</label>
+      <input type="text" name="contactValue" />
+      <br />
+      <select name="contactUse">
+        <option value="home">home</option>
+        <option value="work">work</option>
+        <option value="temp">temporary</option>
+        <option value="old">old</option>
+        <option value="mobile">mobile</option>
+      </select>
+    </>
+  );
+
+  return (
+    <>
+      <h2>Structure definition Settings</h2>
+      <form action="">
+        <label htmlFor="name">Name of profile</label>
+        <input
+          type="text"
+          name="name"
+          value={(name as unknown) as string}
+          onChange={(e) => editName(e)}
+        />
```

Comment:
This HTML tag couple seems to be very redundant.
Maybe creating a generic react component would be useful in this situation.

Also that way, when you'll start to apply style in your components, you would only have to set it in your custom Input label.